src/rtc-to-net/rtc-to-net.ts

```diff
@@ -474,24 +474,21 @@ import logging = require('../logging/logging');
 
     // Sends a packet over the data channel.
     // Invoked when a packet is received over the TCP socket.
-    private sendOnChannel_ = (data:ArrayBuffer) : Promise<void> => {
+    private sendOnChannel_ = (data:ArrayBuffer) : void => {
       this.socketReceivedBytes_ += data.byteLength;
-
-      return this.dataChannel_.send({buffer: data});
+      this.dataChannel_.send({buffer: data});
     }
 
     // Sends a packet over the TCP socket.
     // Invoked when a packet is received over the data channel.
-    private sendOnSocket_ = (data:peerconnection.Data)
-        : Promise<freedom_TcpSocket.WriteInfo> => {
+    private sendOnSocket_ = (data:peerconnection.Data) : void => {
       if (!data.buffer) {
-        return Promise.reject(new Error(
-            'received non-buffer data from datachannel'));
+        throw new Error('received non-buffer data from datachannel');
       }
       this.bytesReceivedFromPeer_.handle(data.buffer.byteLength);
       this.channelReceivedBytes_ += data.buffer.byteLength;
 
-      return this.tcpConnection_.send(data.buffer);
+      this.tcpConnection_.send(data.buffer);
```

Comment:
Reckless calls never raise an exception (that's why they're reckless!).

If the method call throws or rejects in the core environment, it should trigger a freedom log message, so we should still see it in the logs.